scipy-1.0/scipy-optimize.tex

```diff
@@ -124,9 +126,21 @@ \subsubsection*{Nonlinear Optimization}
   \label{tab:minimize}
 \end{table}
 
-Some problems may require the use of global optimizers to find the minimum of the objective function over the set of all parameter values, especially if the objective function possesses many local minima that other solvers could get stuck in.
-\texttt{Brute} performs a grid search over the entire parameter range, with the best grid point undergoing further polishing by another solver. However, this approach becomes increasingly inefficient as the problem dimensionality increases, or if the objective function has many extrema.
-\texttt{Basinhopping} \cite{Wales1997} uses random perturbation of the parameter vector, with local minimization. The acceptance of the new coordinates is dictated by the Metropolis test typically used in Monte Carlo simulations.
-\texttt{differential\textunderscore evolution} \cite{Wormington1999,Storn1997} is a stochastic solver that improves by gradual evolution of a population of candidate solutions. In each iteration trial candidates are generated by combination of candidates from the existing population; if the trial candidates represent an improvement then the population is updated. \texttt{differential\textunderscore evolution} does not require that the objective function be differentiable.
+\paragraph{Global Minimization}
+As \texttt{minimize} only seeks a local minimum, some problems require the use of a global optimization routine to find the minimum of the objective function over the set of all decision variable values, especially if the objective function possesses many local minima that other solvers could get stuck in. % e.g. if there is concern that `minimize` could get stuck in an unacceptable local minimum?
+The \texttt{brute} function performs a grid search over the entire domain, and the best grid point is ``polished'' by a local minimizer. However, this approach becomes prohibitively expensive as the problem dimensionality increases.
+% As far as I understand how brute works, the run time does not depend on the number of extrema, as it only polishes the best one
+% , or if the objective function has many extrema. 
+To address this, the
+\texttt{basinhopping} \cite{Wales1997} function uses random perturbation of the decision variables in conjunction with local minimization. The acceptance of the perturbed coordinates is dictated by the Metropolis test typically used in Monte Carlo simulations. 
+\texttt{differential\textunderscore evolution} \cite{Wormington1999,Storn1997} is a stochastic solver that improves by gradual evolution of a population of candidate solutions. In each iteration, trial candidates are generated by combination of candidates from the existing population; if the trial candidates represent an improvement then the population is updated. 
+% Is the following a unique feature of differential_evolution? If so, state it (although it seems that neither `brute`or `basinhopping` inherently require the a differentiable problem). If not, why mention it? 
```

Comment:
May not be worth mentioning it, but solvers that don't need to use gradients are valuable.

Reply:
Certainly it may be worth mentioning, but I don't think it makes sense to single out only `differential_evolution` if the same is true of all the global optimizers.

If none of the global optimizers inherently require derivatives, then we should say "None of the global optimizers require that the objective function be differentiable." rather than "\texttt{differential\textunderscore evolution} does not require that the objective function be differentiable."

However, maybe your point is really that among the three options you believe that `differential_evolution` is the strongest when derivatives are unavailable. `brute` does not require derivatives to do the grid search, but perhaps it doesn't give a great answer unless a derivative-based local optimizer polishes the solution. `basinhopping` does not inherently require derivatives, but perhaps it performs best when it is used in conjunction with a derivative-based local optimizer. So is the claim that `differential_evolution` works quite well without reliance on a local optimizer that may require derivatives, and that is unique?